android/src/test/kotlin/com/rudderstack/sdk/kotlin/android/plugins/devicemode/IntegrationsManagementPluginTest.kt

```diff
@@ -0,0 +1,294 @@
+            advanceUntilIdle()
+
+            val mockDestinationSdk = integrationPlugin.getUnderlyingInstance() as? MockDestinationSdk
+
+            verify(exactly = 1) { callback.invoke(mockDestinationSdk, ofType(Result.Success::class) as DestinationResult) }
+        }
+
+    @Test
+    fun `given an integration, when a callback is registered after sourceConfig is fetched and integration is added, then it is called after successful initialisation`() =
+        runTest {
+            val callback = mockk<(Any?, DestinationResult) -> Unit>(relaxed = true)
+
+            plugin.setup(mockAnalytics)
+            plugin.addIntegration(integrationPlugin)
+            mockAnalytics.sourceConfigState.dispatch(SourceConfig.UpdateAction(sourceConfigWithCorrectApiKey))
+            plugin.onDestinationReady(integrationPlugin, callback)
+            advanceUntilIdle()
+
+            val mockDestinationSdk = integrationPlugin.getUnderlyingInstance() as? MockDestinationSdk
+
+            verify(exactly = 1) { callback.invoke(mockDestinationSdk, ofType(Result.Success::class) as DestinationResult) }
+        }
+
+    @Test
+    fun `given an integration and multiple callbacks, when they are registered, then each of them is called after successful initialisation`() =
+        runTest {
+            val callback1 = mockk<(Any?, DestinationResult) -> Unit>(relaxed = true)
+            val callback2 = mockk<(Any?, DestinationResult) -> Unit>(relaxed = true)
+
+            plugin.setup(mockAnalytics)
+            plugin.addIntegration(integrationPlugin)
+            mockAnalytics.sourceConfigState.dispatch(SourceConfig.UpdateAction(sourceConfigWithCorrectApiKey))
+            plugin.onDestinationReady(integrationPlugin, callback1)
+            plugin.onDestinationReady(integrationPlugin, callback2)
+            advanceUntilIdle()
+
+            val mockDestinationSdk = integrationPlugin.getUnderlyingInstance() as? MockDestinationSdk
+
+            verify(exactly = 1) { callback1.invoke(mockDestinationSdk, ofType(Result.Success::class) as DestinationResult) }
+            verify(exactly = 1) { callback2.invoke(mockDestinationSdk, ofType(Result.Success::class) as DestinationResult) }
+        }
+
+    @Test
+    fun `given an integration which will fail to initialise and a callback, when it is registered, then it is called with a failure exception after the initialisation attempt`() =
+        runTest {
+            val callback = mockk<(Any?, DestinationResult) -> Unit>(relaxed = true)
+
+            plugin.setup(mockAnalytics)
+            plugin.addIntegration(integrationPlugin)
+            mockAnalytics.sourceConfigState.dispatch(SourceConfig.UpdateAction(sourceConfigWithIncorrectApiKey))
+            plugin.onDestinationReady(integrationPlugin, callback)
+            advanceUntilIdle()
+
+            verify(exactly = 1) { callback.invoke(null, ofType(Result.Failure::class) as DestinationResult) }
+        }
```

Comment:
I think this test case could be improved!

```suggestion
    fun `given an integration which will fail to initialise, when a callback is registered, then it is called with a failure exception after the initialisation attempt`() =
        runTest {
            // Arrange
            val callback = mockk<(Any?, DestinationResult) -> Unit>(relaxed = true)
            plugin.setup(mockAnalytics)
            plugin.addIntegration(integrationPlugin)
            
            // Act
            plugin.onDestinationReady(integrationPlugin, callback)
            mockAnalytics.sourceConfigState.dispatch(SourceConfig.UpdateAction(sourceConfigWithIncorrectApiKey))
            advanceUntilIdle()

            // Assert
            verify(exactly = 1) { callback.invoke(null, ofType(Result.Failure::class) as DestinationResult) }
        }
```

Reply:
Try to follow the assert(or setup), act and assert pattern. 

Currently, it seems the arrange and act phase is mixed up a bit. In future try to follow this pattern for each test.